tests/core/mempool/test_mempool_manager.py

```diff
@@ -251,5 +260,45 @@ async def get_coin_record(_: bytes32) -> Optional[CoinRecord]:
 
     mempool_manager = await instantiate_mempool_manager(get_coin_record)
     conditions = [[ConditionOpcode.CREATE_COIN, IDENTITY_PUZZLE_HASH, 1]]
-    _, _, result = await generate_and_add_spendbundle(mempool_manager, conditions)
+    _, _, result = await generate_and_add_spendbundle(mempool_manager, TEST_COIN, conditions)
     assert result == (None, MempoolInclusionStatus.FAILED, Err.UNKNOWN_UNSPENT)
+
+
+@pytest.mark.asyncio
+async def test_sb_twice_with_eligible_coin_and_different_spends_order() -> None:
+    async def get_coin_record(coin_id: bytes32) -> Optional[CoinRecord]:
+        test_coin_records = {
+            TEST_COIN_ID: TEST_COIN_RECORD,
+            TEST_COIN_ID2: TEST_COIN_RECORD2,
+            TEST_COIN_ID3: TEST_COIN_RECORD3,
+        }
+        return test_coin_records.get(coin_id)
+
+    mempool_manager = await instantiate_mempool_manager(get_coin_record)
+
+    sb1_conditions = [
+        [ConditionOpcode.CREATE_COIN, IDENTITY_PUZZLE_HASH, 1],
+        [ConditionOpcode.CREATE_COIN, IDENTITY_PUZZLE_HASH, 2],
+    ]
+    sb1 = spend_bundle_from_conditions(TEST_COIN, sb1_conditions)
+    sb2_conditions = [
+        [ConditionOpcode.CREATE_COIN, IDENTITY_PUZZLE_HASH, 3],
+        [ConditionOpcode.AGG_SIG_UNSAFE, G1Element(), IDENTITY_PUZZLE_HASH],
+    ]
+    sb2 = spend_bundle_from_conditions(TEST_COIN2, sb2_conditions)
+    sb3_conditions = [[ConditionOpcode.AGG_SIG_UNSAFE, G1Element(), IDENTITY_PUZZLE_HASH]]
+    sb3 = spend_bundle_from_conditions(TEST_COIN3, sb3_conditions)
+    sb = SpendBundle.aggregate([sb1, sb2, sb3])
+    sb_name = sb.name()
+    reordered_sb = SpendBundle.aggregate([sb3, sb1, sb2])
+    reordered_sb_name = reordered_sb.name()
+    assert mempool_manager.get_spendbundle(sb_name) is None
+    assert mempool_manager.get_spendbundle(reordered_sb_name) is None
+    result = await add_spendbundle(mempool_manager, sb, sb_name)
+    assert result == (13091510, MempoolInclusionStatus.SUCCESS, None)
+    assert mempool_manager.get_spendbundle(sb_name) == sb
+    assert mempool_manager.get_spendbundle(reordered_sb_name) is None
+    result = await add_spendbundle(mempool_manager, reordered_sb, reordered_sb_name)
+    assert result == (13091510, MempoolInclusionStatus.PENDING, Err.MEMPOOL_CONFLICT)
```

Comment:
can you add a comment explaining why we should expect `MEMPOOL_CONFLICT` here?

Reply:
Done.